core/controller/src/main/scala/whisk/core/controller/Triggers.scala

```diff
@@ -390,46 +300,130 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
+    val actionUrl = baseControllerPath / rule.action.path.root.asString / "actions"
+
+    val actionPath = {
+      rule.action.path.relativePath.map { pkg =>
+        (Path.SingleSlash + pkg.namespace) / rule.action.name.asString
+      } getOrElse {
+        Path.SingleSlash + rule.action.name.asString
+      }
+    }.toString
+
+    val request = HttpRequest(
+      method = POST,
+      uri = url.withPath(actionUrl + actionPath),
+      headers = List(Authorization(BasicHttpCredentials(user.authkey.uuid.asString, user.authkey.key.asString))),
+      entity = HttpEntity(MediaTypes.`application/json`, args.compactPrint))
+
+    Http().singleRequest(request)
+  }
+
+  /**
+   * Create JSON object containing the pertinent rule activation details.
+   * {
+   *   "rule": "my-rule",
+   *   "action": "my-action",
+   *   "statusCode": 0,
+   *   "status": "success",
+   *   "activationId": "...",                              // either this field, ...
+   *   "error": "The requested resource does not exist."   // ... or this field will be present
+   * }
+   *
+   * @param statusCode one of ActivationResponse values
+   * @param ruleName the name of the rule that was activated
+   * @param actionName the name of the action activated by the rule
+   * @param actionActivationId the activation id, if there is one
+   * @param errorMsg the rror messages otherwise
+   * @return JsObject as formatted above
    */
   private def ruleResult(statusCode: Int,
-                         ruleName: String,
-                         actionName: String,
-                         actionActivationId: Option[String] = None,
+                         ruleName: FullyQualifiedEntityName,
+                         actionName: FullyQualifiedEntityName,
+                         actionActivationId: Option[JsValue] = None,
                          errorMsg: Option[String] = None): JsObject = {
-    val objMap: Map[String, JsValue] = Map(
-      "rule" -> JsString(ruleName),
-      "action" -> JsString(actionName),
+    JsObject(
+      "rule" -> JsString(ruleName.asString),
+      "action" -> JsString(actionName.asString),
       "statusCode" -> JsNumber(statusCode),
-      "success" -> JsBoolean(statusCode == ActivationResponse.Success)) ++ {
-      actionActivationId map { id =>
-        Seq("activationId" -> JsString(id))
-      } getOrElse Seq()
-    } ++ {
-      errorMsg map { err =>
-        Seq("error" -> JsString(err))
-      } getOrElse Seq()
-    }
-
-    // Final rule result looks like
-    // {
-    //   "rule": "my-rule",
-    //   "action": "my-action",
-    //   "statusCode": 0,
-    //   "status": "success",
-    //   "activationId": "90c84e7b33c84ceb884e7b33c8ecebf6",  // Optional
-    //   "error": "The requested resource does not exist."  // Optional
-    // }
-    JsObject(objMap)
+      "success" -> JsBoolean(statusCode == ActivationResponse.Success),
+      "activationId" -> actionActivationId.toJson,
```

Comment:
note: the unit tests (or integration tests) are not tight enough and don't enforce a schema on the response; the activationId and error properties here will emit as `null` if not present which changes the behavior relative to the previous commit.

I suggest a unit test which checks the expected schema exactly; this can be done for the intergration test as well; parse the array to json and check for equality.